src/sync/browser.js

```diff
@@ -0,0 +1,105 @@
+import PushManagerFactory from './PushManager';
+import { PUSH_DISCONNECT, PUSH_CONNECT } from './constants';
+import FullProducerFactory from '../producer';
+import PartialProducerFactory from '../producer/browser/Partial';
+import { matching } from '../utils/key/factory';
+import { forOwn } from '../utils/lang';
+import logFactory from '../utils/logger';
+const log = logFactory('splitio-sync:sync-manager');
+
+/**
+ * Factory of sync manager for browser
+ *
+ * @param context main client context
+ */
+export default function BrowserSyncManagerFactory(mainContext) {
+
+  const contexts = {};
+  let pushManager;
+
+  function startPolling() {
+    log.info('PUSH down or disconnected. Starting periodic fetch of data.');
+    forOwn(contexts, function (context) {
+      const producer = context.get(context.constants.PRODUCER, true);
+      if (producer && !producer.isRunning())
+        producer.start(true); // `fetchers` are scheduled but not called immediately
+    });
+  }
+
+  function stopPollingAndSyncAll() {
+    log.info('PUSH (re)connected. Syncing and stopping periodic fetch of data.');
+    // if polling, stop
+    forOwn(contexts, function (context) {
+      const producer = context.get(context.constants.PRODUCER, true);
+      if (producer && producer.isRunning())
+        producer.stop();
+    });
+    syncAll();
+  }
+
+  function syncAll() {
+    // fetch splits and segments
+    const mainProducer = mainContext.get(mainContext.constants.PRODUCER, true);
+    mainProducer && mainProducer.synchronizeSplits();
+    forOwn(contexts, function (context) {
+      const producer = context.get(context.constants.PRODUCER, true);
+      producer && producer.synchronizeMySegments();
+    });
+  }
+
+  function shared(context, isSharedClient = true) {
+    const producer = isSharedClient ? PartialProducerFactory(context) : FullProducerFactory(context);
+    context.put(context.constants.PRODUCER, producer);
```

Comment:
Put the variables towards the top and assignments after unless it makes sense execution~wise or you're defining blocks using spaces/comments. Any of those work.